src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup/Cmdlets/Restore/RestoreAzureRMRecoveryServicesBackupItem.cs

```diff
@@ -88,27 +150,49 @@ public override void ExecuteCmdlet()
                 ResourceIdentifier resourceIdentifier = new ResourceIdentifier(VaultId);
                 string vaultName = resourceIdentifier.ResourceName;
                 string resourceGroupName = resourceIdentifier.ResourceGroupName;
+                Dictionary<Enum, object> providerParameters = new Dictionary<Enum, object>();
 
-                GenericResource storageAccountResource = GetStorageAccountResource();
+                GenericResource storageAccountResource = GetStorageAccountResource(StorageAccountName,
+                    StorageAccountResourceGroupName);
                 WriteDebug(string.Format("StorageId = {0}", storageAccountResource.Id));
+                providerParameters.Add(RestoreBackupItemParams.StorageAccountId, storageAccountResource.Id);
+                providerParameters.Add(RestoreBackupItemParams.StorageAccountLocation, storageAccountResource.Location);
+                providerParameters.Add(RestoreBackupItemParams.StorageAccountType, storageAccountResource.Type);
+
+                if (ParameterSetName == AzureFileParameterSet && TargetStorageAccountName != null)
```

Comment:
Since this validation is specific to the provider, can we move this to the Provider layer?

Reply:
This validation is using a common function in cmdlet layer

Reply:
Can we move the common function to the provider common layer?